### PR TITLE
Fix for heap memory leak on recoverable errors and new points

### PR DESCRIPTION
**Pull request by nikolay-t**
This pull request fixes issue described by https://github.com/influxdata/influxdb-java/issues/541.

The issue was caused by failure to increase count of points when they were merged into existing batchPoints object. The merge changes total number of points, but this was not counted.

The idea of the fix is to recalculate local counter and evict old batches not only on ordinary batchQueue update but also in case of successful merge.

This PR also contains change in unit tests: new test was added to ensure that total number of points in batchQueue lists equals to local counter variable, as well as that total number of points never exceeds user defined limit.

### CODE REVIEW

**Comment by nikolay-t** [comment]
@dubsky @lxhoan @bednar @fmachado Please review.

**Comment by nikolay-t** [comment]
CI failure looks like related to https://github.com/carlossg/docker-maven/issues/90. Any idea how to fix it? Will try to test it in my fork.

**Comment by ninja-** [comment]
@majst01 why is this not merged?

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ env:
   - MAVEN_JAVA_VERSION=3-jdk-11-slim INFLUXDB_VERSION=1.3
   - MAVEN_JAVA_VERSION=3-jdk-11-slim INFLUXDB_VERSION=1.2
   - MAVEN_JAVA_VERSION=3-jdk-11-slim INFLUXDB_VERSION=1.1
-  - MAVEN_JAVA_VERSION=3-jdk-8-slim INFLUXDB_VERSION=1.7 SNAPSHOT=Y
+    # switched to alpine due to "slim" issue https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=911925
+  - MAVEN_JAVA_VERSION=3-jdk-8-alpine INFLUXDB_VERSION=1.7 SNAPSHOT=Y
```
Comment (majst01) [inline .travis.yml]: This change is not needed anymore, the bug was fixed with a newer jacoco plugin.

Comment (nikolay-t) [reply .travis.yml]: This change was reverted.

**Comment by nikolay-t** [comment]
@majst01 Travis CI seems to fail due to unrelated problem: 
```
[ERROR]     Unresolveable build extension: Plugin org.sonatype.plugins:nexus-staging-maven-plugin:1.6.8 or one of its dependencies could not be resolved: The following artifacts could not be resolved: org.sonatype.nexus.maven:nexus-common:jar:1.6.8, com.google.guava:guava:jar:14.0.1: Could not transfer artifact org.sonatype.nexus.maven:nexus-common:jar:1.6.8 from/to central (https://repo.maven.apache.org/maven2): Connect to repo.maven.apache.org:443 [repo.maven.apache.org/151.101.184.215] failed: Connection timed out (Connection timed out) -> [Help 2]
```
Should I repeat the checks somehow? I don't think it is related to my changes.

**Pull request merged**